pkg/cloud/metadata_ec2.go

```diff
@@ -34,11 +36,19 @@ func EC2MetadataInstanceInfo(svc EC2Metadata) (*Metadata, error) {
 	}
 
 	if len(doc.Region) == 0 {
-		return nil, fmt.Errorf("could not get valid EC2 region")
+		if len(regionFromSession) != 0 && util.IsSBE(regionFromSession) {
+			doc.Region = regionFromSession
+		} else {
+			return nil, fmt.Errorf("could not get valid EC2 region")
+		}
 	}
 
 	if len(doc.AvailabilityZone) == 0 {
-		return nil, fmt.Errorf("could not get valid EC2 availability zone")
+		if len(regionFromSession) != 0 && util.IsSBE(regionFromSession) {
+			doc.AvailabilityZone = regionFromSession
```

Comment:
Why are we putting region name into AZ? Is there no way to determine the AZ?

Reply:
The availability zone for SBE is same as the region and there is no other way to determine it since snow metadata service doesn't provide any information about it.